sunpy/net/vso/vso.py

```diff
@@ -168,6 +171,21 @@ def iter_errors(response):
         if not hasattr(prov_item, 'record') or not prov_item.record:
             yield prov_item
 
+def check_connection(url):
+    try:
+        return requests.get(url).status_code == 200
+    except (socket.error, socket.timeout) as e:
+        warnings.warn("Connection failed with error {}. \n Retrying with different url and port.".format(e))
```

Comment:
For completeness I would make this return `False` rather than implicit `None` as it then always returns `bool`